lib/GenomeFileUtil/core/GenbankToGenome.py

```diff
@@ -62,75 +65,152 @@ def __init__(self, config):
-            'taxon_wsname': self.cfg.raw['taxon-workspace-name'],
-            'taxon_lookup_obj_name': self.cfg.raw['taxon-lookup-object-name'],
+            'taxon_wsname': cfg.raw['taxon-workspace-name'],
+            'taxon_lookup_obj_name': cfg.raw['taxon-lookup-object-name'],
 
-            'ontology_wsname': self.cfg.raw['ontology-workspace-name'],
-            'ontology_GO_obj_name': self.cfg.raw['ontology-gene-ontology-obj-name'],
-            'ontology_PO_obj_name': self.cfg.raw['ontology-plant-ontology-obj-name'],
+            'ontology_wsname': cfg.raw['ontology-workspace-name'],
+            'ontology_GO_obj_name': cfg.raw['ontology-gene-ontology-obj-name'],
+            'ontology_PO_obj_name': cfg.raw['ontology-plant-ontology-obj-name'],
 
             'release': None,
             'genetic_code': 11,
             'generate_ids_if_needed': 0,
             'metadata': {}
         }
 
-    @property
-    def messages(self):
-        return "\n".join(self._messages)
-
-    def refactored_import(self, ctx, params):
-        # 1) validate parameters and extract defaults
-        self.validate_params(params)
-
-        # 2) construct the input directory staging area
-        input_directory = self.stage_input(params)
-
-        # 3) update default params
-        self.default_params.update(params)
-        params = self.default_params
-        self.generate_parents = params.get('generate_missing_genes')
-        self.generate_ids = params.get('generate_ids_if_needed')
-        if params.get('genetic_code'):
-            self.code_table = params['genetic_code']
-
-        # 4) Do the upload
-        files = self._find_input_files(input_directory)
-        consolidated_file = self._join_files_skip_empty_lines(files)
-        genome = self.parse_genbank(consolidated_file, params)
-        if params.get('genetic_code'):
-            genome["genetic_code"] = params['genetic_code']
-
-        result = self.gi.save_one_genome({
-            'workspace': params['workspace_name'],
-            'name': params['genome_name'],
-            'data': genome,
-            "meta": params['metadata'],
-        })
-        ref = f"{result['info'][6]}/{result['info'][0]}/{result['info'][4]}"
-        logging.info(f"Genome saved to {ref}")
 
-        # 5) clear the temp directory
-        shutil.rmtree(input_directory)
-
-        # 6) return the result
-        info = result['info']
-        details = {
-            'genome_ref': ref,
-            'genome_info': info
-        }
+class GenbankToGenome:
+    def __init__(self, config):
+        self.cfg = config
+        self.gi = GenomeInterface(config)
+        self.dfu = DataFileUtil(config.callbackURL)
+        self.aUtil = AssemblyUtil(config.callbackURL)
+        self.ws = Workspace(config.workspaceURL)
+        self.re_api_url = config.re_api_url
+        yml_text = open('/kb/module/kbase.yml').read()
+        self.version = re.search("module-version:\n\W+(.+)\n", yml_text).group(1)
```

Comment:
I realize you didn't write this, but this is bonkers - the version is available in the impl file that calls the constructor here, so why not just pass it in from there or add it to `config`?

Reply:
kbase.yml has a version of 0.11.6
Impl has a version of 0.11.0

Ideally they should be consistent?

Reply:
kb-sdk compile will fix that

Reply:
Which happens on catalog registraion

Reply:
https://github.com/kbaseapps/GenomeFileUtil/blob/master/Makefile#L20-L25

Reply:
also the fix in kb-sdk compile?

Reply:
I'm not sure what you mean exactly - the purpose of kb-sdk compile is to take the spec file and generate the service and impl files. When it generates the impl file (saving any already existing code) it uses the version from the kbase.yml file.

Reply:
run `make compile` and you'll see what I mean

Reply:
👍

Reply:
:+1: